src/app/asset-page/asset-page.component.ts

```diff
@@ -765,15 +765,15 @@ export class AssetPage implements OnInit, OnDestroy {
         else {
             for (let col of this.collections) {
                 // Private/Personal Collection
-                if (col.type === '6') {
+                if (col.type === '6' && col.name === value) {
                     return ['/pcollection', col.id]
                 }
                 // Public Collection
-                else if (col.type === '5') {
+                else if (col.type === '5' && col.name === value) {
                     asset.publicDownload = true
                     return ['/collection', col.id]
                 }
-                else {
+                else if (col.name === value) {
```

Comment:
@rehanabbasi  This all looks good. Can you look into the use of the `link[]` array here. I slightly remember writing this, but it looks as if `link` isn't used anymore, and that this line, (Line#777) should `return` the result. I just noticed this in glancing over the PR, so feel free to change this up a bit more if you feel it would be better.